src/conversions.jl

```diff
@@ -194,9 +194,111 @@ function tttcg(jd1, jd2)
     date, date1
 end
 
+"""
+    tdbtt(jd1, jd2, dtr)
+
+Transform a two-part Julia date from `TDB` to `TT`.
+
+# Example
+
+```jldoctest
+julia> tdb = Epoch{TDB}(2.4578265e6, 0.30440190993249416)
+2017-03-14T07:18:20.325 TDB
+julia> AstronomicalTime.Epochs.tdbtt(tdb.jd1, tdb.jd2, AstronomicalTime.Epochs.deltatr(tdb))
+(2.4578265e6, 0.30440190993249416)
+```
+"""
+function tdbtt(jd1, jd2, dtr)
+    dtrd = dtr / SECONDS_PER_DAY
+    if jd1 > jd2
+        date = jd1
+        date1 = jd2 - dtrd
+    else
+        date = jd1 - dtrd
+        date1 = jd2
+    end
+    date, date1
+end
+
+
+function dtdb(jd1, jd2, ut, elong, u, v)
+
+    t = ((jd1 - DJ00) + jd2) / DJM
+    # Convert UT to local solar time in radians.
+     tsol = mod(ut, 1.0) * 2π  + elong
+
+    # FUNDAMENTAL ARGUMENTS:  Simon et al. 1994.
+    # Combine time argument (millennia) with deg/arcsec factor.
+     w = t / 3600.0
+    # Sun Mean Longitude.
+     elsun = mod(280.46645683 + 1296027711.03429 * w, 360.0) * DEG_2R
+    # Sun Mean Anomaly.
+     emsun = mod(357.52910918 + 1295965810.481 * w, 360.0) * DEG_2R
+    # Mean Elongation of Moon from Sun.
+     d = mod(297.85019547 + 16029616012.090 * w, 360.0) * DEG_2R
+    # Mean Longitude of Jupiter.
+     elj = mod(34.35151874 + 109306899.89453 * w, 360.0) * DEG_2R
+    # Mean Longitude of Saturn.
+     els = mod(50.07744430 + 44046398.47038 * w, 360.0) * DEG_2R
+    # TOPOCENTRIC TERMS:  Moyer 1981 and Murray 1983.
+     wt =   +  0.00029e-10 * u * sin(tsol + elsun - els)
+            +  0.00100e-10 * u * sin(tsol - 2.0 * emsun)
+            +  0.00133e-10 * u * sin(tsol - d)
+            +  0.00133e-10 * u * sin(tsol + elsun - elj)
+            -  0.00229e-10 * u * sin(tsol + 2.0 * elsun + emsun)
+            -  0.02200e-10 * v * cos(elsun + emsun)
+            +  0.05312e-10 * u * sin(tsol - emsun)
+            -  0.13677e-10 * u * sin(tsol + 2.0 * elsun)
+            -  1.31840e-10 * v * cos(elsun)
+            +  3.17679e-10 * u * sin(tsol)
+    # =====================
+    # Fairhead et al. model
+    # =====================
+
+    # T**0
+     w0 = 0
+     for j in 474:-1:1
+        w0 += fairhd[j][1] * sin(fairhd[j][2] * t + fairhd[j][3])
+    end
+    # T**1
+     w1 = 0
+     for j in 679:-1:1
```

Comment:
The lower bound of this loop and of the following ones are wrong.

Reply:
The loops all work on non-overlapping segements of the coefficients (or would if they were correct 😉). Thus, the first step to simplify this would be to split coefficients into five separate arrays.